Makefile

```diff
@@ -0,0 +1,72 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+SHELL = /bin/bash
+# We assume an active virtualenv for development
+PYENV_REGEX = .pyenv/shims
+PY_BIN = python3
+# https://github.com/pypa/pip/issues/5599
+PIP_WRAPPER = $(PY_BIN) -m pip
+export PY38 = $(shell jq -r '.python_versions.PY38' .ci/variables.json)
+VENV_NAME ?= .venv
+VENV_ACTIVATE_FILE = $(VENV_NAME)/bin/activate
+VENV_ACTIVATE = . $(VENV_ACTIVATE_FILE)
+VEPYTHON = $(VENV_NAME)/bin/$(PY_BIN)
+VEPYLINT = $(VENV_NAME)/bin/pylint
+PYENV_ERROR = "\033[0;31mIMPORTANT\033[0m: Please install pyenv.\n"
+PYENV_PREREQ_HELP = "\033[0;31mIMPORTANT\033[0m: please type \033[0;31mpyenv init\033[0m, follow the instructions there and restart your terminal before proceeding any further.\n"
+VE_MISSING_HELP = "\033[0;31mIMPORTANT\033[0m: Couldn't find $(PWD)/$(VENV_NAME); have you executed make venv-create?\033[0m\n"
+export TRACK_DIRS = $(shell find -E . -regex ".*/track\.json"|sed -r 's|/[^/]+$$||' | sort -u)
+
+prereq:
+	pyenv install --skip-existing $(PY38)
+	pyenv local $(PY38)
+	@# Ensure all Python versions are registered for this project
+	@ jq -r '.python_versions | [.[] | tostring] | join("\n")' .ci/variables.json > .python-version
+	-@ printf $(PYENV_PREREQ_HELP)
+
+venv-create:
```

Comment:
Do we want venv-create to be executed as part of make install, similarly to rally Makefile?